### PR TITLE
Added new mission: Infiltration Commando

### PR DESCRIPTION
**Pull request by OcelotWalrus**
NOTE: Requires #1055's merge, as it requires the difficulties addon.

### CODE REVIEW

File: mods/hv/maps/infiltration-commando/infiltration-commando.lua [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+	if Difficulty == "easy" then
+		NavalReinforcements = Reinforcements.Reinforce(Human, { "patrolboat", "patrolboat", "submarine", "submarine" }, NavalReinforcementPath)
+	end
+end
+
+SendAirstrikeWaves = function()
+	-- Send airstrike waves at different locations, and different delays.
+	-- They start at 02mins20secs and are all spaced by 01mins35secs
+	-- There are 18 waves, with 2 targeting the player's base
+	Trigger.AfterDelay(8500, function()
```
Comment (Mailaender) [inline mods/hv/maps/infiltration-commando/infiltration-commando.lua:55]:
```suggestion
	Trigger.AfterDelay(DateTime.Minutes(2) + DateTime.Seconds(30), function()
```

Comment (OcelotWalrus) [reply mods/hv/maps/infiltration-commando/infiltration-commando.lua:55]: Sure, it's better for human-reading. I'll do that throughout the script.

**Comment by Dzierzan** [comment]
The game crashes upon building harbor:
image

[lua.log](https://github.com/user-attachments/files/16364639/lua.log)

**Comment by OcelotWalrus** [comment]
That's because you rename mediumboat to mercenaryboat.

**Comment by Dzierzan** [comment]
I haven't played it after your recent commit, but you sure want to use Merc Boat instead of Torpedo one?

**Comment by Dzierzan** [comment]
Alright, I've got more feedback:

1. Disable "Money tank", it should be only used in MP, unless there's clear purpose for that in SP mission (like protection or something)

2. It's a huge chokepoint for AI. you might wanna redo that part.
image

Actually, this base has only one entry, no wonder AI had a huge problem here, considering Storage buildings are in another base. I'd make more entries to that base, this should help.
image

3. Another huge chokepoint:
image
Either give here some gates or redo these parts to gave AI more space.

4. I dunno why you prefer dark maps so much, but they're not so very fun when it comes to visibility. Please, make it brighter.

5. I am not sure if disabling Banshee was a good idea. I mean, I just spammed Copters instead... like a huge spam. And a huge snowball of units is always effective against anything anyway.

**Comment by OcelotWalrus** [comment]
If you're having trouble completing the mission, you can just check this replay: https://easyupload.io/4ocvgl

**Comment by Dzierzan** [comment]
Honestly, the way you placed Generators doesn't look too good. I recommend such placement:
image

I'd would also add some patrols... maybe some ships and aircraft, what do you think?

**Comment by OcelotWalrus** [comment]
> Honestly, the way you placed Generators doesn't look too good.

Yeah these placements look better. Thank for pointing that out.

> I'd would also add some patrols... maybe some ships and aircraft, what do you think?

There are actually already air patrols, that regenerate if they're destroyed. I guess I could add naval patrols too.

**Comment by Dzierzan** [comment]
I'd make more land here, it's a chokepoint due to that bunkers' placement.
image

**Comment by Dzierzan** [comment]
I'd maybe rework that part a bit:
image

**Comment by Dzierzan** [comment]
Anyway... I think AI spams maybe way too many units, I had to fight against huge army of naval units and I totally forgot about the damn colony. I'd reduce AI naval squad if possible.

**Comment by OcelotWalrus** [comment]
> I'd make more land here, it's a chokepoint due to that bunkers' placement.
>
> Maybe allign those turrets better?
>
> I'd maybe rework that part a bit:

Yes, indeed.

> Anyway... I think AI spams maybe way too many units, I had to fight against huge army of naval units and I totally forgot about the damn colony. I'd reduce AI naval squad if possible.

Why saying "maybe" if you say after "way too much"? Again, it's supposed to be a hard mission. But I'll reduce the time space between enemy naval squad production: with the current time, it actually produces all the time. But I won't reduce the squad size. Again, the AI won't produce naval units unless the player builds and harbor or has saved the colony. So, just focus on saving the colony first.

**Comment by Dzierzan** [comment]
So you're saying it's a puzzle mission where building Harbor is not recommended because it aggros AI, despite you encourage the player in building the harbor (secondary objective, lots of water).

Reduce the squad size, you know the solution of your puzzle, potential player won't.

**Comment by OcelotWalrus** [comment]
I won't say it's a puzzle mission, but one that'll require you to play several times to succeed. I and @Rampoina played the mission and dealing with the naval units while capturing the radar/attack the main base wasn't a problem. If you find it too hard, you can just set the difficulty to easier then. You just gotta produce naval a keep them close to your base to defend.

**Comment by Dzierzan** [comment]
Ohh believe me, I am not some sunday player, I know my stuff. But I clearly noticed an issue where people might have a problem.

I played it on normal and the amount of units AI had felt like very hard, you had to counter attack with yet equal spam of your units.

> but one that'll require you to play several times to succeed

Sorry, but the way you wrote it does sound like a puzzle mission where you want player to fail in order to do stuff in a correct order.

I do understand the mission is meant to be hard, that's actually great, but there's a thin line between hard mission and frustrating mission. We wanna find golden mean here.

**Comment by OcelotWalrus** [comment]
I'll reduce the squad size then.
I'd like to know why you fell like it's a "puzzle mission".

**Comment by Rampoina** [comment]
I didn't feel the naval units were too much of a problem. I kind of agree with @Dzierzan regarding the feel of the mission though.

The intended way to play it is to *rush* to the enemy Radar Dome and take it with a tech, it gives very very little margin for error. If you don't know how to play it you will lose until you figure out you have to do that, and it's like a "puzzle" like @Dzierzan was saying.

The first few times I played it I just didn't understand that and was trying to just beat the AI and take down the power plants, you did improve this a bit by showing the radar dome and telling the player to capture it in-game.

The power plants are shown to the player but are of little importance to stop the airstrikes (they do delay them a bit but it's almost impossible to stop it before losing more than 25%).

> Honestly, the way you placed Generators doesn't look too good. I recommend such placement:

This change actually made it even harder. When the power plants are clumped you can at least destroy a few with a single howitzer salvo but now you can't.

My suggestion is to make destroying the power plants a viable strategy to stall the air strikes. Either massively reduce the amount of generators the AI has or script it directly so that the player has to keep destroying power plants.

Otherwise don't show the power plants to the players at all because it's a red herring.

**Comment by OcelotWalrus** [comment]
Thanks for that review!
What I'm gonna try to do is make so that the more yuruki power plants are destroyed, the more the airstrikes occur. I'm also gonna edit the Howitzer weapon to make its damage apply to a larger area, so it's more efficient with the new power plant pattern.

**Comment by OcelotWalrus** [comment]
Should I add some announcer? Like for the reminders and everything.

File: mods/hv/maps/infiltration-commando/weapons.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# License: CC-BY-SA-4.0
+
+HowitzerOvercharged:
```
Comment (Dzierzan) [inline mods/hv/maps/infiltration-commando/weapons.yaml:3]: IMO you shouldn't be modifing Howitzer, unit stats should be rather on paar with MP ones. This makes stuff inconsistent between missions.

Comment (OcelotWalrus) [reply mods/hv/maps/infiltration-commando/weapons.yaml:3]: The problem is that without modifying it, it's not powerful enough for the mission.

File: mods/hv/languages/rules/en.ftl [hunk 3]
```diff
@@ -84,6 +84,9 @@ resource-iron = Base Metal Deposit
 bot-rogue-ai =
    .name = Rogue AI
 
+bot-campaign-ai =
+   .name = Campaign AI
```
Comment (Mailaender) [inline mods/hv/languages/rules/en.ftl:88]: This is already part of
https://github.com/OpenHV/OpenHV/blob/0445cb22de695c6d8d72a7aaf56663766d9fc090/mods/hv/languages/campaign/en.ftl#L1-L3

File: mods/hv/maps/infiltration-commando/map.yaml [hunk 4]
```diff
@@ -0,0 +1,6171 @@
...
+	Actor2038: barrier
+		Owner: Yuruki Industries
+		Location: 95,60
+	Actor2039: barrier
+		Owner: Yuruki Industries
+		Location: 84,45
+
+Rules: hv|rules/campaign.yaml, rules.yaml
+
+Translations: hv|languages/difficulty/en.ftl, hv|languages/lua/en.ftl, en.ftl
```
Comment (Mailaender) [inline mods/hv/maps/infiltration-commando/map.yaml:6167]: You are missing the `hv|languages/campaign/en.ftl`

File: mods/hv/languages/lua/en.ftl [hunk 5]
```diff
@@ -67,3 +67,23 @@ rescue-scientist = Rescue the scientist.
...
+yuruki-harbor-production-detected = Detected Yuruki naval production
+power-plants-positions-revealed = Power plant outpost localized
+percent-still-alive = % of colony still online
+save-colony-easy = Save at least 55% of the colony.
+save-colony-normal = Save at least 70% of the colony.
+save-colony-hard = Save at least 80% of the colony.
+infiltrate-radar = Capture Yuruki radar outpost.
+annihilate-harbor-facilities = Neutralize Yuruki harbor facilities.
+build-radar = Build a Radar.
+build-howitzer = Build an Howitzer.
```
Comment (Mailaender) [inline mods/hv/languages/lua/en.ftl:82]:
```suggestion
build-howitzer = Build a Howitzer.
```

File: mods/hv/languages/lua/en.ftl [hunk 6]
```diff
@@ -67,3 +67,23 @@ rescue-scientist = Rescue the scientist.
...
+build-harbor = Build an Harbor.
```
Comment (Mailaender) [inline mods/hv/languages/lua/en.ftl:83]:
```suggestion
build-harbor = Build a Harbor.
```

File: mods/hv/maps/infiltration-commando/weapons.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+			Light: 60
+			Heavy: 60
+			Wood: 40
+	Warhead@Incendiary: TreeDamage
+		Damage: 25000
+		Spread: 850
+		Percentage: 40
+	Warhead@Effect: CreateEffect
+		Explosions: large
+		ImpactSounds: explosion07.ogg, explosion08.ogg
```
Comment (Dzierzan) [inline mods/hv/maps/infiltration-commando/weapons.yaml:33]: I am still against making howitzer stronger.

File: mods/hv/maps/infiltration-commando/weapons.yaml (hunk 2 above)
Comment (Mailaender) [reply mods/hv/maps/infiltration-commando/weapons.yaml:3]: This is still not solved.

File: mods/hv/maps/infiltration-commando/rules.yaml [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+		Palette: cloud
+		Speed: 20, 30
+	LuaScript:
+		Scripts: common|scripts/utils.lua, hv|scripts/campaign.lua, infiltration-commando.lua
+	MissionData:
+		Briefing: A Synapol mining colony is being threatened by a neraby continental airstrike Yuruki outpost. The basement and its surroundings are extremely well protected by automated defenses, and ground, naval and air patrols. The outpost is split in different parts, which are alimented by many different power plant outposts.\n\nFind the base's communication quarters in the northeastern continent and capture their radar dome to stop the airstrike waves. You must save the colony's buildings from incoming airstrike waves. During your mission, target the Yuruki power plant outposts to delay the airstrikes and power outage their automatic defense systems.\n\nAlternatively, you can disembark into the continent throughout naval or air transports, and destroy the 4 basements in the headquarters' center, near the Yuruki mothership.\n\nNuke Launch Commands and Drop Pods Power are disabled during this mission. Howitzers will have a higher charge rate and damage area during this mission.
```
Comment (Mailaender) [inline mods/hv/maps/infiltration-commando/rules.yaml:39]: Extract this string into `.ftl` file.

File: mods/hv/maps/infiltration-commando/infiltration-commando.lua [hunk 9]
```diff
@@ -0,0 +1,392 @@
...
+	DestroyHarborFacilities = AddPrimaryObjective(Human, "annihilate-harbor-facilities")
+	BuildRadarObjective = AddSecondaryObjective(Human, "build-radar")
+	BuildHowitzerObjective = AddSecondaryObjective(Human, "build-howitzer")
+	BuildHarborObjective = AddSecondaryObjective(Human, "build-harbor")
+	AlternativeBreakIn = AddSecondaryObjective(Human, "alternative-break-in")
+
+	-- Update patrol troops units to match the difficulty selected
+	if Difficulty == "easy" then
+		NavalPatrol = { "lightboat", "lightboat", "torpedoboat", "lightningboat" }
+		GroundPatrol = { "mgpod", "mgpod", "electricpod", "mbt", "artillery", "apc", "bike" }
```
Comment (Mailaender) [inline mods/hv/maps/infiltration-commando/infiltration-commando.lua:375]: I renamed the pods again so this crashes now. Sorry.

**Review by Mailaender — CHANGES_REQUESTED** [review]

File: mods/hv/maps/infiltration-commando/weapons.yaml (hunk 7 above)
Comment (OcelotWalrus) [reply mods/hv/maps/infiltration-commando/weapons.yaml:33]: I'm gonna test if the mission still works if its armament is the default one.

**Comment by OcelotWalrus** [comment]
Seems like mission briefings are not supported. I checked and tried everything (map ftl, common ftl, lua ftl...) and it only displays `infiltration-commando-briefing`.

**Comment by OcelotWalrus** [comment]
Note that your will to make the howitzer have the same armament as default makes the howitzer not really useful as it was designed to be for the mission. It worked well with the older power plant pattern, but not now, if its armament spread is not enhanced.

**Comment by Mailaender** [comment]
Don't make skirmish weapons inconsistent in single-player. It is irritating and bad design.

**Comment by Rampoina** [comment]
> Don't make skirmish weapons inconsistent in single-player. It is irritating and bad design.

I mean you can just justify it in mission by saying it's a special powerful howitzer. (and maybe make it bigger or something)

**Comment by Dzierzan** [comment]
Then it's a backdoor for people for making always custom stuff in missions. And this stuff has to be remembered when changes are made for Skirmish/MP. Hence we're against that.

File: mods/hv/maps/infiltration-commando/rules.yaml [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+	AttackOrderPower:
+		ChargeInterval: 2100
+
+SILO:
+	Buildable:
+		Prerequisites: ~disabled
+
+GENERATOR:
+	Power:
+		Amount: 200
```
Comment (Dzierzan) [inline mods/hv/maps/infiltration-commando/rules.yaml:89]: No, please make it default 120.

**Review by Dzierzan — CHANGES_REQUESTED** [review]

**Comment by Dzierzan** [comment]
BTW the oggs you added are nice and here comes a problem. You've got now two announcers. You could have asked me to do them, then it would be consistent.

**Comment by OcelotWalrus** [comment]
I can make myself the mission announcer, as I plan on reworking every mission.

**Comment by Rampoina** [comment]
> Then it's a backdoor for people for making always custom stuff in missions. And this stuff has to be remembered when changes are made for Skirmish/MP. Hence we're against that.

That's kinda the point of campaigns and missions to have special stuff that isn't in the normal multiplayer because it's unique or too strong or whatever else.

**Comment by Dzierzan** [comment]
> I can make myself the mission announcer, as I plan on reworking every mission. And to be honest I don't really like your announcer, it's too much modified and doesn't sound natural in my opinion.

You can like or not my announcer voice, but the issue with yours is that you've got a conflict with mine now. And it's not meant to sound natural. Think of CnC3 Nod announcer.

Anyway, I'd add custom announcer voices in another PR, when it's established what we all want.

> That's kinda the point of campaigns and missions to have special stuff that isn't in the normal multiplayer because it's unique or too strong or whatever else.

But we don't wanna later keep remembering about custom stuff in missions, it's a small project without huge manpower.

**Comment by OcelotWalrus** [comment]
> you've got a conflict with mine now

There ain't any conflict.

> you've got a conflict with mine now

I don't see any conflict here.

File: mods/hv/maps/infiltration-commando/rules.yaml [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+		Intensity: 0.90
+	CloudSpawner:
+		Image: cloud
+		Sequences: cloud1, cloud1f, cloud2, cloud2f, cloud3, cloud3f
+		Palette: cloud
+		Speed: 20, 30
+	LuaScript:
+		Scripts: common|scripts/utils.lua, hv|scripts/campaign.lua, infiltration-commando.lua
+	MissionData:
+		Briefing: infiltration-commando-briefing
```
Comment (Mailaender) [inline mods/hv/maps/infiltration-commando/rules.yaml:39]: Ah sorry, I was wrong. https://github.com/OpenRA/OpenRA/pull/21526 is still not merged.

**Pull request merged**